### PR TITLE
cot editor cli add

### PR DESCRIPTION
**Pull request by yavko**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Adds Cot Editor Cli Support
**What is the current behavior? (You can also link to an open issue here)**
No cot cli support
**What is the new behavior (if this is a feature change)?**
cot cli support
**Additional info:**
3rd contribution

### CODE REVIEW

File: dev/cot.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/* 
+Made by Yavko 🥺
+CotEditor is great I hope you use this!
+*/
+const completionSpec: Fig.Spec = {
+  name: "cot",
+  icon: "https://coteditor.com/img/appicon/512@2x.png",
```
Comment (fedeci) [inline dev/cot.ts:7]: Isn't this image a bit too large?

File: dev/cot.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      name: ["-g", "--background"],
+    },
+    {
+      description: "Create a new blank document.",
+      name: ["-n", "--new"],
+    },
+    {
+      description: "Jump to specific line in opened document.",
+      name: ["-l", "--line"],
+      insertValue: "--line '{cursor}'",
```
Comment (fedeci) [inline dev/cot.ts:36]:
```suggestion
      args: {
        name: "line",
      },
```

File: dev/cot.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    },
+    {
+      description: "Jump to specific column in opened document.",
+      name: ["-c", "--column"],
+      insertValue: "--column '{cursor}'",
```
Comment (fedeci) [inline dev/cot.ts:41]:
```suggestion
      args: {
        name: "column",
      },
```

File: dev/cot.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+const completionSpec: Fig.Spec = {
+  name: "cot",
+  icon: "https://coteditor.com/img/appicon/512@2x.png",
+  description: "Command-line utility for CotEditor.",
+  args: {
+    template: "filepaths",
+  },
+  options: [
+    {
+      description: "Show this help message and exit.",
```
Comment (fedeci) [inline dev/cot.ts:14]: nit:
```suggestion
      description: "Show help message and exit.",
```

Comment (yavko) [reply dev/cot.ts:14]: Its like that in the cli help text so i did it like that

File: dev/cot.ts (hunk 1 above)
Comment (yavko) [reply dev/cot.ts:7]: whats a good size?

**Comment by yavko** [comment]
@fedeci I fixed the changes!

**Review by Milo123459 — COMMENTED** [review]
Should be the description of Cot, not the completion spec afaik

File: dev/cot.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
+/* 
+Made by Yavko 🥺
+CotEditor is great I hope you use this!
+*/
+const completionSpec: Fig.Spec = {
+  name: "cot",
+  icon: "view-source:https://coteditor.com/favicon@2x.png",
```
Comment (fedeci) [inline dev/cot.ts:7]:
```suggestion
  icon: "https://coteditor.com/favicon@2x.png",
```

**Review by fedeci — APPROVED** [review]

**Comment by fedeci** [comment]
@Milo123459 what do you mean?

**Comment by Milo123459** [comment]
> @Milo123459 what do you mean?

Was resolved in the Discord, mb

**Comment by yavko** [comment]
> Should be the description of Cot, not the completion spec afaik
cot is a optional module
you can install
for cot editor
It is offical btw

**Comment by yavko** [comment]
how can i make the icon use the fig protocol

**Comment by fedeci** [comment]
[These](https://fig.io/docs/reference/suggestion/icon-api) are the docs. I would keep it as is for the spec, however.

File: dev/cot.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
+/* 
+Made by Yavko 🥺
+CotEditor is great I hope you use this!
+*/
```
Comment (Nutlope) [inline dev/cot.ts:4]: Can you replace this with "// Made by Yavko" please? We're gonna standerdize on this so if someone wants to add their name to a spec that they made, they can just add "Made by BLANK" at the top of the spec.

Comment (yavko) [reply dev/cot.ts:4]: ok!

Comment (yavko) [reply dev/cot.ts]: done

**Comment by Nutlope** [comment]
LGTM!

**Comment by yavko** [comment]
Thanks for the merge!

> #3 contribution!

**Pull request merged**